File: Pilha/Conversor.c
#include <stdio.h>
#include <stdlib.h>
#include "Pilha.h"

int main()
{
    int valor, resto;
    struct Pilha p;
    create(&p);

    printf("Digite um valor ai velhinho...");
    scanf("%d", &valor);

    while (valor != 0)
    {
        resto = valor % 2;
        push(&p, resto);
        valor = valor / 2;
    }

    while (!isEmpty(p))
    {
        resto = pop(&p);
        printf("%d ",resto);
    }
    printf("\n\nFim do Programa!\n\n");
    return 0;
}

File: Pilha/Conversor.java
import java.util.Scanner;

public class Conversor {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.println("Digite algum numero ai velho");
        int numero = sc.nextInt();
        sc.close();
        Pilha p = new Pilha();

        int resto;

        // fase 1 - empilhamento dos restos
        while (numero != 0) {
            resto = numero % 2; // pega resto da divisao do numero por 2
            p.push(resto); // ponho na pilha
            numero = numero / 2; // divide o numero por 2
        }

        // fase 2 - exibicao dos restos
        while (!p.isEmpty()) {
            resto = p.pop(); // remove da pilha o cara q esta no topo
            System.out.print(resto);
        }
    }
}
